FileShareHelper.cs

```diff
@@ -0,0 +1,97 @@
+﻿using Microsoft.Azure.Storage;
+using Microsoft.Azure.Storage.File;
+using NLog.Common;
+using System;
+using System.Threading.Tasks;
+
+namespace NLog.Extensions.AzureFileSystem
+{
+    public static class FileShareHelper
+    {
+        public static CloudFileShare InitializeFileShare(IAzureFileSystemTarget target, CloudFileShare fileShare, string folderName, string fileName)
+        {
+            return AsyncHelper.RunSync(async () => await InitializeFileShareAsync(target, fileShare, folderName, fileName));
+        }
+
+        public static async Task<CloudFileShare> InitializeFileShareAsync(IAzureFileSystemTarget target, CloudFileShare fileShare, string folderName, string fileName)
+        {
+            if (fileShare == null)
+            {
+                fileShare = await CreateStorageAccountFromConnectionStringAsync(target)
+                    .ConfigureAwait(false);
+            }
+
+            folderName = folderName.Replace(@"\", @"/");
+
+            var rootDir = fileShare.GetRootDirectoryReference();
+
+            // Get a reference to the directory.
+            var folder = rootDir.GetDirectoryReference(folderName);
+            var folderExists = await folder.ExistsAsync().ConfigureAwait(false);
+
+            if (!folderExists)
+            {
+                var levels = folderName.Split(new char[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
+                var currentLevel = string.Empty;
```

Comment:
I think that you can directly initialise the currentLevel with levels[0] - the root folder

Reply:
I changed the loop to start with the first level. It feels unsafe but the condition above that I added which checks for null or empty and assumption that root level exists 100% should be fine.